### PR TITLE
AWS Pre-signed url

### PR DESCRIPTION
**Pull request by knockknockyoo**
1. Created AWS Domain
2. Created S3 Service
- Pre-signed url

### CODE REVIEW

**Review by n2iw — CHANGES_REQUESTED** [review]
See other comment for details.

File: services/s3-service.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class S3Service {
+
+    constructor(s3Auth) {
+        this.s3Auth = s3Auth;
+    }
+
+    async createPreSignedURL(bucketName, submissionID, fileName) {
```
[the remark this answers is not in the document]
Comment (knockknockyoo) [reply services/s3-service.js]: Replaced with a prefix parameter

File: services/s3-service.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    constructor() {
+        this.s3 = new AWS.S3();
+    }
+
+    async createPreSignedURL(bucketName, prefix, fileName) {
+        try {
+            const params = {
+                Bucket: bucketName,
+                Key: `${prefix}`,
```
Comment (n2iw) [inline services/s3-service.js:12]: I think this should be **`${prefix}/${fileName}`**

**Review by n2iw — CHANGES_REQUESTED** [review]

File: services/s3-service.js (hunk 2 above)
Comment (knockknockyoo) [reply services/s3-service.js]: updated the object key

**Review by n2iw — APPROVED** [review]
LGTM

**Pull request merged**